### ISSUE TITLE
Inconsistent Parameter Type for Floats

### ISSUE THREAD

**Issue by matekelemen** [KratosMultiphysics/Kratos#10302]
## TL;DR
```Parameters::IsDouble``` returns ```false``` for integers, but ```Parameters::GetDouble``` parses the value without exceptions anyway.

## Details
The question is how the following piece of code should behave:
```py
parameters = KratosMultiphysics.Parameters("""{"a_numeric_value" : 42}""")
parameters["a_numeric_value"].IsDouble()
parameters["a_numeric_value"].GetDouble()
```
Currently, ```IsDouble()``` returns ```false```. While this guarantees that exactly one of ```Parameters::Is...()``` returns ```true``` for any valid input (== no ambiguity about the type of any value), it makes parameter checking/validation a bit more messy (see de422c0f4c5719ae5a20649829f7b4640cbb0c5b).

However, ```GetDouble()``` yields ```42.0``` without issues, which is a bit surprising. In general, if an object is not of type ```T```, it shouldn't be possible to parse it as ```T``` directly.

## Opinion
I don't think the type system imposed on JSON structures should be too strict but it should absolutely be consistent, so I'd change ```IsDouble()``` to return ```true``` for integers as well. I know this behaviour comes directly from ```nlohmann::json```, but I don't see why ```Parameters``` should behave identically to it. This only implies the following change:

```kratos_parameters.cpp```
```diff
bool Parameters::IsDouble() const
{
-    return mpValue->is_number_float();
+    return mpValue->is_number();
}
```

The alternative would that ```IsDouble()``` yields ```false``` and ```GetDouble()``` throws an exception, which isn't a great idea in my opinion.

**Comment by loumalouomega** [comment]
Huuuum, I mostly agree

**Comment by loumalouomega** [comment]
> Huuuum, I mostly agree

But I think we decided to use the number directly to avoid problems of people forgetting to punt the ' .0' at the end

**Comment by RiccardoRossi** [comment]
sorry i did not see this. To me the current behaviour is the correct one

**Comment by RiccardoRossi** [comment]
i mean, you should be able to give an input an integer, test if the type of input is integer, but if you want cast it to a double (same way you would by a type cast)
